3-enrich/emr-etl-runner/lib/snowplow-emr-etl-runner/emr.rb

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2012-2018 Snowplow Analytics Ltd. All rights reserved.
+#
+# This program is licensed to you under the Apache License Version 2.0,
+# and you may not use this file except in compliance with the Apache License Version 2.0.
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+
+# Author::    Joshua Beemster (mailto:support@snowplowanalytics.com)
+# Copyright:: Copyright (c) 2012-2018 Snowplow Analytics Ltd
+# License::   Apache License Version 2.0
+
+require 'aws-sdk-emr'
+require 'contracts'
+require 'pathname'
+require 'uri'
+
+module Snowplow
+  module EmrEtlRunner
+    module EMR
+
+      include Contracts
+
+      # Attempts to find an active EMR JobFlow with a given name
+      #
+      # Parameters:
+      # +client+:: EMR client
+      # +name+:: EMR cluster name
+      def get_emr_jobflow_id(client, name)
+        get_emr_jobflow_id_impl(client, name)
+      end
+
+      private
+
+      def get_emr_jobflow_id_impl(client, name)
```

Comment:
What was the reason for second private function?

Reply:
Following the same code pattern as @BenFradet implemented in his S3 class - happy to drop this back to a single public function.